tests/monitoring/monitoring.go

```diff
@@ -157,17 +157,36 @@ var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", func() {
 		})
 	})
 
-	Context("Alerts runbooks", func() {
-		It("Should have available URLs", func() {
-			alerts, err := getAlerts(virtClient)
+	Context("Kubevirt alert rules", func() {
+
+		It("should have all the requried annotations", func() {
+			monv1 := virtClient.PrometheusClient().MonitoringV1()
+			prometheusRule, err := monv1.PrometheusRules(flags.KubeVirtInstallNamespace).Get(context.Background(), components.KUBEVIRT_PROMETHEUS_RULE_NAME, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
-			for _, alert := range alerts {
-				Expect(alert.Annotations).ToNot(BeNil())
-				url, ok := alert.Annotations["runbook_url"]
-				Expect(ok).To(BeTrue())
-				resp, err := http.Head(string(url))
-				Expect(err).ToNot(HaveOccurred())
-				Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+			for _, group := range prometheusRule.Spec.Groups {
+				for _, rule := range group.Rules {
+					if len(rule.Alert) > 0 {
+						Expect(rule.Annotations).ToNot(BeNil())
+						checkForRunbookURL(rule)
+						checkForSummary(rule)
+					}
+				}
+			}
+		})
+
+		It("should have all the requried labels", func() {
+			monv1 := virtClient.PrometheusClient().MonitoringV1()
+			prometheusRule, err := monv1.PrometheusRules(flags.KubeVirtInstallNamespace).Get(context.Background(), components.KUBEVIRT_PROMETHEUS_RULE_NAME, metav1.GetOptions{})
```

Comment:
Both `monv1` and `prometheusRule, err` are used in both `It` specs. Can we use `BeforeEach` to avoid code duplications?

Reply:
Done